test/performance/metrics/runtime.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"k8s.io/apimachinery/pkg/labels"
+	deploymentinformer "knative.dev/pkg/injection/informers/kubeinformers/appsv1/deployment"
+	netv1alpha1 "knative.dev/serving/pkg/apis/networking/v1alpha1"
+	sksinformer "knative.dev/serving/pkg/client/injection/informers/networking/v1alpha1/serverlessservice"
+
+	// Mysteriously required to support GCP auth (required by k8s libs).
+	// Apparently just importing it is enough. @_@ side effects @_@. https://github.com/kubernetes/client-go/issues/242
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+// DeploymentStatus is a struct that wraps the status of a deployment.
+type DeploymentStatus struct {
+	DesiredReplicas int32
+	ReadyReplicas   int32
+	// Time is the time when the status is fetched
+	Time time.Time
+}
+
+// FetchDeploymentStatus creates a channel that can return the up-to-date DeploymentStatus periodically.
+func FetchDeploymentStatus(
+	ctx context.Context, namespace string, selector labels.Selector,
+	duration time.Duration, stop <-chan struct{},
```

Comment:
`context.Context` has a built in `Done()` channel, it seems we should be using that if we can.  I think you want `ctx, cancel := context.WithCancel(ctx)`, which will make `ctx.Done()` close when `cancel()` is called.

Reply:
Done, thanks for the suggestion!